### PR TITLE
Add try except

### PR DESCRIPTION
**Pull request by shira263**
### Added 
Added new exception KustoNetworkError
Added try except in client and cloud_settings

### CODE REVIEW

**Review by AsafMah — CHANGES_REQUESTED** [review]

File: azure-kusto-data/tests/kusto_client_common.py [hunk 1]
```diff
@@ -89,6 +89,8 @@ def raise_for_status(self):
             file_name = "zero_results.json"
         elif "PrimaryResultName" in kwargs["json"]["csl"]:
             file_name = "null_values.json"
+        elif "raiseNetowrk" in kwargs["json"]["csl"]:
+            file_name = "noFile.json"
```
[the remark this answers is not in the document]
Comment (shira263) [reply azure-kusto-data/tests/kusto_client_common.py:93]: Can you give an example for an actual call that fails? 
Why isn't it suitable to use a file that doesn't exist? this is exactly what we want to validate, a case which the "try" is not working and stepping into except.

File: azure-kusto-data/tests/test_token_providers.py [hunk 2]
```diff
@@ -278,6 +280,19 @@ def test_app_key_provider():
         else:
             print(" *** Skipped App Id & Key Provider Test ***")
 
+    @staticmethod
+    def test_app_key_provider_whenURLnotValid():
```
[the remark this answers is not in the document]
Comment (shira263) [reply azure-kusto-data/tests/test_token_providers.py:284]: Do you mean a test to test the changes in client? I added the tests for the client in test_kusto_client

File: azure-kusto-data/azure/kusto/data/exceptions.py [hunk 3]
```diff
@@ -125,6 +125,13 @@ def get_api_error(self) -> OneApiError:
         return self.error
 
 
+class KustoNetworkError(KustoServiceError):
```
[the remark this answers is not in the document]
Comment (shira263) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: Can you give an example to other exception that I can use as an reference? I used as a reference other exceptions that looks like what I implemented and does not contain endpoint or inner exception.

File: azure-kusto-data/tests/kusto_client_common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AsafMah) [reply azure-kusto-data/tests/kusto_client_common.py:93]: 1. Spelling - "raiseNetwork"
2. `throw` an exception that is suitable for this case.

File: azure-kusto-data/azure/kusto/data/exceptions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (AsafMah) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: See further down the file for more examples.
I want you to know how to write and use classes for the next assignment.

File: azure-kusto-data/tests/test_token_providers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AsafMah) [reply azure-kusto-data/tests/test_token_providers.py:284]: What you currently have:
1. Unit test for the client
2. Integration test for cloud info

I want you to add an actual end-to-end test for the client.

Look at e2e.py (in azure-kusto-data).

Create a test that: 
1. Makes a client.
2. Uses an address that doesn't exist
3. Verify the exception being thrown.

Both sync and async.

File: azure-kusto-data/azure/kusto/data/exceptions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shira263) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: Added the endpoint. The inner already exist when using 'from e' in client.py. Why do you prefer to pass it as a parameter instead?

File: azure-kusto-data/tests/test_token_providers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (shira263) [reply azure-kusto-data/tests/test_token_providers.py:284]: Talked offline and agreed not to do.

File: azure-kusto-data/azure/kusto/data/exceptions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (AsafMah) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: We discussed this, right?

[the remark this answers is not in the document]
Comment (shira263) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: If I recall correctly, you said we are not used to pass it as a parameter, but after I did some research, I saw that it is more common to use "from e"

[the remark this answers is not in the document]
Comment (AsafMah) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: I'll clarify:

1. Currently you only put endpoint as part of the message, I want it to be a field so people can get it without parsing the error message.
2. Add the client request id as a member too

**Review by AsafMah — CHANGES_REQUESTED** [review]
see reply

File: azure-kusto-data/azure/kusto/data/exceptions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shira263) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: 1. But we do we use "from exception" in other parts of the code. 
Do you mean you want it to be a member for future use? 
2. Is it also relevant when we raise the error from _cloud_settings?

[the remark this answers is not in the document]
Comment (AsafMah) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: I don't need the inner exception as a member, only what I said.
Client request id isn't available in cloud settings

[the remark this answers is not in the document]
Comment (shira263) [reply azure-kusto-data/azure/kusto/data/exceptions.py:128]: 1. Let's talk about it offline.
2. This is why I am asking

File: azure-kusto-data/azure/kusto/data/exceptions.py [hunk 4]
```diff
@@ -125,6 +125,17 @@ def get_api_error(self) -> OneApiError:
...
+    """Raised when a Kusto client fails to connect to network."""
+
+    def __init__(self, endpoint: str, client_request_id=None):
+        super().__init__(
+            "Failed to process network request for the endpoint: " + endpoint + "" if client_request_id is None else "Client Request ID:" + client_request_id
```
Comment (AsafMah) [inline azure-kusto-data/azure/kusto/data/exceptions.py:133]:
```suggestion
            "Failed to process network request for the endpoint: " + endpoint + ( "" if client_request_id is None else ("Client Request ID:" + client_request_id) )
```

**Review by AsafMah — DISMISSED** [review]
Fix the suggestion and you can merge

**Review by AsafMah — APPROVED** [review]

**Pull request merged**